cognite/pygen/_query/executor.py

```diff
@@ -159,13 +163,13 @@ def iterate(
                 if e.code == 408:
                     # Too big query, try to reduce the limit
                     if self._reduce_max_batch_limit():
+                        new_limit = status.max_retrieve_batch_limit
+                        warnings.warn(
+                            f"Query is too large, reducing batch size to {new_limit:,}, and trying again",
+                            QueryReducingBatchSize,
+                            stacklevel=2,
+                        )
```

Comment:
Bugfix 2: Moved warning to correct location where it will be printed on batch size reduction.